jax/_src/numpy/lax_numpy.py

```diff
@@ -975,15 +988,40 @@ def interp(x: ArrayLike, xp: ArrayLike, fp: ArrayLike,
   dx0 = lax.abs(dx) <= epsilon  # Prevent NaN gradients when `dx` is small.
   f = where(dx0, fp_arr[i - 1], fp_arr[i - 1] + (delta / where(dx0, 1, dx)) * df)
 
-  left_arr: ArrayLike = fp_arr[0] if left is None else left
-  right_arr: ArrayLike = fp_arr[-1] if right is None else right
+  if not extrapolate_left:
+    assert not isinstance(left, str)
+    left_arr: ArrayLike = fp_arr[0] if left is None else left
+    if period is None:
+      f = where(x_arr < xp_arr[0], left_arr, f)
+  if not extrapolate_right:
+    assert not isinstance(right, str)
+    right_arr: ArrayLike = fp_arr[-1] if right is None else right
+    if period is None:
+      f = where(x_arr > xp_arr[-1], right_arr, f)
 
-  if period is None:
-    f = where(x_arr < xp_arr[0], left_arr, f)
-    f = where(x_arr > xp_arr[-1], right_arr, f)
   return f
 
 
+@util._wraps(np.interp,
+  lax_description=_dedent("""
+    In addition to constant interpolation supported by NumPy, jnp.interp also
+    supports left='extrapolate' and right='extrpolate' to indicate linear
+    extrpolation instead."""))
+def interp(x: ArrayLike, xp: ArrayLike, fp: ArrayLike,
+           left: Union[ArrayLike, str, None] = None,
+           right: Union[ArrayLike, str, None] = None,
+           period: Optional[ArrayLike] = None) -> Array:
+  static_argnames = []
+  if isinstance(left, str) or left is None:
```

Comment:
This broke previously (only on old versions of NumPy!) because I didn't include `or left is None` for deciding to use a static argname.